src/frontend/src/btc/schedulers/btc-wallet.scheduler.ts

```diff
@@ -123,16 +124,18 @@ export class BtcWalletScheduler implements Scheduler<PostMessageDataRequestBtc>
 		assertNonNullish(btcAddress, 'No BTC address provided to get BTC transactions.');
 
 		const balance = await this.loadBtcBalance({ identity, bitcoinNetwork });
-		const newTransactions = data?.shouldFetchTransactions
-			? await this.loadBtcTransactions({ btcAddress })
+		const { newTransactions, latestBitcoinBlockHeight } = data?.shouldFetchTransactions
+			? await this.loadBtcTransactionsData({ btcAddress })
+			: { newTransactions: [], latestBitcoinBlockHeight: undefined };
+
+		// TODO: handle the case when tx data is available but latestBitcoinBlockHeight is undefined
+		const uncertifiedTransactions = nonNullish(latestBitcoinBlockHeight)
```

Comment:
should it still post a message is there are no new transactions? meaning, does the balance change if there are no new transactions?

same question above where you check for `shouldFetchTransactions`: if it shouldn't fetch transactions, do you still want to post a message?

Reply:
1. Since we are fetching balance from one source (IC) and txs from the other one (blockchain API), there is a chance that they not 100% synced (hence balance can get an update even with no new transactions). Therefore, it makes sense to post a message (at least, for now), and try to solve this race condition problem globally. 
2. `shouldFetchTransactions` is `true` only for mainnet. It means that we have to post a message for testnet and regtest even if we don't have any transactions at all.

Reply:
thank you for the explanation!